frontend/public/components/utils/service-level.tsx

```diff
@@ -0,0 +1,318 @@
+import { getOCMLink } from '../../module/k8s';
+import { k8sGet } from '@console/dynamic-plugin-sdk/src/api/core-api';
+import { SecretModel } from '../../models';
+import { ExternalLink, FieldLevelHelp } from './index';
+import { RootState } from '../../redux';
+
+const useServiceLevelText = (level: string): string => {
+  const { t } = useTranslation();
+  const levels = {
+    Premium: t('public~Premium'),
+    Standard: t('public~Standard'),
+    'Self-Support': t('public~Self-support'),
+    Eval: t('public~Self-support, 60 day trial'),
+    None: t('public~None'), // Eval has ended
+    Unknown: t('public~Unknown'), // Not officially returned from the API, but is used when no results are returned from the API
+  };
+
+  if (!level) {
+    return levels.Unknown;
+  }
+  return levels[level] || level;
+};
+
+const showServiceLevel = (clusterID: string) =>
+  clusterID &&
+  window.SERVER_FLAGS.branding !== 'okd' &&
+  window.SERVER_FLAGS.branding !== 'azure' &&
+  window.SERVER_FLAGS.branding !== 'dedicated';
+
+const TrialDaysLeft: React.FC<{
+  level: string;
+  trialDaysLeft: number | null;
+  label?: boolean;
+}> = ({ level, trialDaysLeft, label }) => {
+  const { t } = useTranslation();
+
+  if ((level !== 'Eval' && level !== 'None') || (level === 'Eval' && trialDaysLeft === null)) {
+    return null;
+  }
+
+  let variant: 'warning' | 'danger' = 'warning';
+
+  let alertText = t('public~{{count}} day remaining', { count: trialDaysLeft });
+  if (trialDaysLeft === 0) {
+    alertText = t('public~< 1 day remaining');
+  }
+
+  if (trialDaysLeft < 0 || level === 'None') {
+    variant = 'danger';
+    alertText = t('public~Trial expired');
+  }
+
+  if (label) {
+    return (
+      <div>
+        <Label
+          color={variant === 'warning' ? 'orange' : 'red'}
+          icon={
+            variant === 'warning' ? <YellowExclamationTriangleIcon /> : <RedExclamationCircleIcon />
+          }
+        >
+          {alertText}
+        </Label>
+      </div>
+    );
+  }
+  return (
+    <Alert
+      className="co-alert-inline--no-bold-title"
+      variant={variant}
+      isInline
+      isPlain
+      title={alertText}
+    />
+  );
+};
+
+const useLoadServiceLevel = (): [boolean, boolean, (clusterID: string) => void] => {
+  const dispatch = useDispatch();
+  const [loadingSecret, setLoadingSecret] = React.useState(false);
+  const [loadingServiceLevel, setLoadingServiceLevel] = React.useState(false);
+
+  let hasSecretAccess = false;
+
+  const loadServiceLevel = (clusterID: string): void => {
+    if (!showServiceLevel(clusterID)) {
+      setLoadingSecret(false);
+      setLoadingServiceLevel(false);
+      dispatch(UIActions.setServiceLevel(null, null, clusterID, null, hasSecretAccess));
+      return;
+    }
+    setLoadingSecret(true);
+    k8sGet({ model: SecretModel, name: 'pull-secret', ns: 'openshift-config' })
+      .then((response) => {
+        // @ts-ignore  Data is not recognized as part of response.
```

Comment:
Verify that this is still an issue and still needs to be commented out.  It seems that there would be a TS error here and not other places